src/main/java/com/clivelee/crawler/EmailCrawler.java

```diff
@@ -0,0 +1,49 @@
+package com.clivelee.crawler;
+
+import com.clivelee.utils.TextFinder;
+import edu.uci.ics.crawler4j.crawler.Page;
+import edu.uci.ics.crawler4j.crawler.WebCrawler;
+import edu.uci.ics.crawler4j.parser.HtmlParseData;
+import edu.uci.ics.crawler4j.url.WebURL;
+
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+
+final class EmailCrawler extends WebCrawler {
+    private final static Pattern FILTERS = Pattern.compile(".*(\\.(css|js|html))$");
+
+    private String domainName;
+    private Set<String> emails;
+
+    EmailCrawler(String domainName, Set<String> emails) {
+        super();
+        this.domainName = domainName.toLowerCase();
+        this.emails = emails;
+    }
+
+    @Override
+    public boolean shouldVisit(Page referringPage, WebURL url) {
+        String href = url.getURL().toLowerCase();
+        return !FILTERS.matcher(href).matches()
```

Comment:
This is overly restrictive and as such misses many pages linked from the start point of web.mit.edu.  Can you see the mistake and fix it?

Reply:
hmm i suppose if the url has query parameters or fragments regex wouldn't match. or there may be other file types we might want as well, like htm.